plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveColumnProjectionInfo.java

```diff
@@ -108,4 +113,14 @@ public static String generatePartialName(List<String> dereferenceNames)
                 .map(name -> "#" + name)
                 .collect(Collectors.joining());
     }
+
+    public long getRetainedSizeInBytes()
+    {
+        return INSTANCE_SIZE
+                + estimatedSizeOf(dereferenceIndices, SizeOf::sizeOf)
+                + estimatedSizeOf(dereferenceNames, SizeOf::estimatedSizeOf)
+                + hiveType.getRetainedSizeInBytes()
+                // type is not accounted for as the instances are cached (by TypeRegistry) and shared
```

Comment:
It is based on implementation detail of other component. Also cache is size bound so technically it is not sure that we will get cached instance here (though very probable).
How big the type objects are? Maybe it is better to still account them here to be on the safe side.

Reply:
The memory accounting primitives for `Type` is not very complex to implement, but also non trivial. To implement proper memory accounting the `TypeSignature` class and `TypeSignatureParameter` class will have to provide memory accounting capabilities as well. In case of `TypeSignatureParameter` which stores the value as `Object` in adds additional implementation complexity, as based on the type of parameter the value of a parameter can be of 4 different types that would all have to be accounted for.

From one perspective the complexity added is not that high. But from the practical perspective I'm not sure if it is worth it.

As I already mentioned in the comment the types are cached by `TypeRegistry`. Many simple types are singletons (such as `BigintType`). The only cache in `TypeRegistry` that is bounded is `parametricTypeCache`. But it accommodates 1000 types, and in reality it is rather unlikely to overflow (as it would mean that at the current moment the system processes queries over more than 1000 columns that all have distinct types). But even so - types are usually reused across all splits as the column handles are usually resolved only once before the split enumeration starts. Thus accounting memory for splits is more likely than not to cause double counting for the memory used by those types.